lib/Sema/CSDiag.cpp

```diff
@@ -5421,8 +5431,92 @@ bool FailureDiagnosis::visitApplyExpr(ApplyExpr *callExpr) {
     // Type check the function subexpression to resolve a type for it if
     // possible.
     fnExpr = typeCheckChildIndependently(callExpr->getFn());
-    if (!fnExpr)
+    if (!fnExpr) {
+      auto UDE = dyn_cast<UnresolvedDotExpr>(callExpr->getFn());
+      if (!UDE)
+        return true;
+
+      auto baseExpr = UDE->getBase();
+      // Offer fix-it for using `subscript` instead of the operator.
+      // FIXME: This doesn't check generic parameters
+      if (baseExpr && UDE->getName().getBaseName() == "subscript") {
+        auto baseType = CS.getType(baseExpr);
+        // Look up subscript declarations and return if there are none.
+        auto lookup = CS.lookupMember(baseType->getRValueType(),
+                                      DeclName(DeclBaseName::createSubscript()));
+        if (lookup.empty())
+          return true;
+
+        auto argExpr = typeCheckChildIndependently(callExpr->getArg(), Type(),
+                                                   CTP_CallArgument);
+        if (!argExpr)
+          return true;
```

Comment:
This is a bit different from above but I think it should return `CS.TC.Diags.hadAnyError();` because otherwise there is going to be the same result...